eng/cmake/global_compile_options.cmake

```diff
@@ -6,15 +6,22 @@ if(MSVC)
   )
 endif()
 
+if(ADDRESS_SANITIZER)
+  add_compile_options(-fsanitize=address)
+  if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    add_link_options(-fsanitize=address)
+  endif()
+endif()
+
 # Turn on strict compiler flags only for testing to allow better compatability with diverse platforms.
 if(UNIT_TESTING)
   if(MSVC)
     if(WARNINGS_AS_ERRORS)
       set(WARNINGS_AS_ERRORS_FLAG "/WX")
-      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WX")
+      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /WX /ignore:4300 /ignore:4302")
```

Comment:
What end up motivating this change?

Reply:
Please add those reasons for ignoring those warnings as a comment to the cmake file, so anyone looking at the sources in the future can see why (or with a link to what these warnings are).